File: 프로그래머스/Level1/체육복.py
n = 5  # 전체학생 수
lost = [1, 5]  # 체육복을 잃어버린 학생
reserve = [1, 4, 5]  # 여벌의 체육복이 있는 학생


def solution(n, lost, reserve):
    tmp = list(lost)
    for l in tmp:  # 최종적으로 여벌을 갖는 학생
        if l in reserve:
            lost.remove(l)
            reserve.remove(l)
    # print(f"lost: {lost} reserve: {reserve}")
    tmp = list(lost)
    for l in tmp:
        if l-1 in reserve:
            lost.remove(l)
            reserve.remove(l-1)
        elif l+1 in reserve:
            lost.remove(l)
            reserve.remove(l+1)
    answer = n-len(lost)
    print(answer)
    return answer


solution(n, lost, reserve)

File: Javascript/프로그래머스/Level1/체육복.js
function solution(n, lost, reserve) {
    var answer = 0;
    var new_l = [];

    for (const el of lost) {
        if (reserve.includes(el)) {
            let tmp = reserve.indexOf(el);
            reserve.splice(tmp, 1);
        } else {
            new_l.push(el);
        }
    }

    var lostcnt = 0;
    for (const el of new_l) {
        if (reserve.includes(el - 1)) {
            let tmp = reserve.indexOf(el - 1);
            reserve.splice(tmp, 1);
        } else if (reserve.includes(el + 1)) {
            let tmp = reserve.indexOf(el + 1);
            reserve.splice(tmp, 1);
        } else {
            lostcnt += 1;
        }
    }
    answer = n - lostcnt;
    return answer;
}
